lib/src/text_animation_widget.dart

```diff
@@ -33,39 +44,90 @@ class TextAnimationWidget extends StatelessWidget {
+        TextSelection(baseOffset: i, extentOffset: i + 1),
+        boxHeightStyle: BoxHeightStyle.max,
+        boxWidthStyle: BoxWidthStyle.max,
+      );
+
+      if (selectionRects.isNotEmpty) {
+        boxes.add((character: text[i], box: selectionRects.first, index: i));
+      }
+    }
+    final boxSize = textPainter.size;
+    textPainter.dispose();
+    return (boxes: boxes, overallBoxSize: boxSize);
+  }
+
+  Widget renderCharacter(
+      AnimationProperty animationProperty, Movie movie, TextBoxInfo info) {
+    return Opacity(
+      opacity: animationProperty.opacity.fromOrDefault(movie).clamp(0, 1),
+      child: Transform(
+        alignment: animationProperty.transformation
+            .fromOrDefault(movie)
+            .transformAlignment,
+        transform: animationProperty.transformation.fromOrDefault(movie).matrix,
+        child: Text.rich(
+          TextSpan(text: info.character),
+          style: textStyle,
+        ),
+      ),
+    );
+  }
+
   @override
   Widget build(BuildContext context) {
-    return LoopAnimationBuilder(
-      tween: tween,
-      builder: (context, movie, _) {
-        return Text.rich(
-          TextSpan(
-            children: animationInput.groups.indexed.map((char) {
-              final index = char.$1;
-              final value = char.$2;
-
-              final animationProperty = animationProperties.elementAt(index);
-              return WidgetSpan(
-                child: Opacity(
-                  opacity: animationProperty.opacity
-                      .fromOrDefault(movie)
-                      .clamp(0, 1),
-                  child: Transform(
-                    alignment: animationProperty.transformation
-                        .fromOrDefault(movie)
-                        .transformAlignment,
-                    transform: animationProperty.transformation
-                        .fromOrDefault(movie)
-                        .matrix,
-                    child: Text(value, style: textStyle),
-                  ),
-                ),
+    return Align(
+      alignment: textAlign.toAlignment(),
+      child: LayoutBuilder(builder: (context, constraints) {
+        final boxInfo = getCharacterDetails(context, constraints);
+        return SizedBox.fromSize(
+          size: boxInfo.overallBoxSize,
+          child: LoopAnimationBuilder(
+            tween: tween,
+            builder: (context, movie, child) {
+              return Stack(
+                clipBehavior: Clip.none,
+                children: boxInfo.boxes
+                    .mapIndexed(
+                      (i, e) => renderCharacterAnimation(e, movie),
+                    )
+                    .toList(growable: false),
               );
-            }).toList(),
+            },
+            duration: tween.duration,
           ),
         );
-      },
-      duration: tween.duration,
+      }),
+    );
+  }
+
+  Positioned renderCharacterAnimation(TextBoxInfo info, Movie movie) {
```

Comment:
can we rename this?

Reply:
the name is actually correct, but the method body does not indicate it is rendering an animated widget. is the renderCharacter method needed, we could add the widget as a inner widget here, since it is small